qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,312 @@
+
+    @property
+    def min_eigen_solver(self) -> MinimumEigensolver:
+        """Return the minimum eigensolver."""
+        return self._min_eigen_solver
+
+    @min_eigen_solver.setter
+    def min_eigen_solver(self, min_eigen_solver: MinimumEigensolver) -> None:
+        """Set the minimum eigensolver."""
+        if not min_eigen_solver.supports_aux_operators():
+            raise TypeError(
+                f"The provided MinimumEigensolver ({type(min_eigen_solver)}) "
+                "does not support auxiliary operators."
+            )
+        self._min_eigen_solver = min_eigen_solver
+
+    @property
+    def max_vars_per_qubit(self) -> int:
+        """Return the maximum number of variables per qubit."""
+        return self._max_vars_per_qubit
+
+    @max_vars_per_qubit.setter
+    def max_vars_per_qubit(self, max_vars_per_qubit: int) -> None:
+        """Set the maximum number of variables per qubit."""
+        self._max_vars_per_qubit = max_vars_per_qubit
+
+    @property
+    def rounding_scheme(self) -> RoundingScheme:
+        """Return the rounding scheme."""
+        return self._rounding_scheme
+
+    @rounding_scheme.setter
+    def rounding_scheme(self, rounding_scheme: RoundingScheme) -> None:
+        """Set the rounding scheme."""
+        self._rounding_scheme = rounding_scheme
+
+    def get_compatibility_msg(self, problem: QuadraticProgram) -> str:
+        """Checks whether a given problem can be solved with this optimizer.
+
+        Checks whether the given problem is compatible, i.e., whether the problem can be converted
+        to a QUBO, and otherwise, returns a message explaining the incompatibility.
+
+        Args:
+            problem: The optimization problem to check compatibility.
+
+        Returns:
+            A message describing the incompatibility.
+        """
+        return QuadraticProgramToQubo.get_compatibility_msg(problem)
+
+    def solve_relaxed(
+        self,
+        encoding: QuantumRandomAccessEncoding,
+    ) -> tuple[MinimumEigensolverResult, RoundingContext]:
+        """Solve the relaxed Hamiltonian given by the encoding.
+
+        Args:
+            encoding: The ``QuantumRandomAccessEncoding``, which must have already been ``encode()``ed
+                with a ``QuadraticProgram``.
+
+        Returns:
+            The result of the minimum eigensolver, and the rounding context.
+
+        Raises:
+            ValueError: If the encoding has not been encoded with a ``QuadraticProgram``.
+        """
+        if not encoding.frozen:
+            raise ValueError(
+                "The encoding must call ``encode()`` with a ``QuadraticProgram`` before being passed"
+                "to the QuantumRandomAccessOptimizer."
+            )
+
+        # Get the list of operators that correspond to each decision variable.
+        variable_ops = [encoding._term2op(i) for i in range(encoding.num_vars)]
+
+        # Solve the relaxed problem
+        start_time_relaxed = time.time()
```

Comment:
I suggest `timeit.default_timer` instead of `time.time`.
https://docs.python.org/3/library/timeit.html